### PR TITLE
convert featuregate FAQ from google doc to part of the dev-guide

### PR DESCRIPTION
**Pull request by deads2k**

### CODE REVIEW

File: dev-guide/featuresets.md [hunk 1]
```diff
@@ -0,0 +1,468 @@
...
+    with(gateGatewayAPI).
+```
+
+## openshift/your-repo
+```go
+desiredVersion := config.OperatorReleaseVersion
+missingVersion := "0.0.1-snapshot"
+
+// By default, this will exit(0) if the featuregates change
+featureGateAccessor := featuregates.NewFeatureGateAccess(
```
Comment (ibihim) [inline dev-guide/featuresets.md:172]: A code box with the imports would be nice.

**Review by ibihim — COMMENTED** [review]
/lgtm

**Comment by jan--f** [comment]
Other files (though not all) have a metadata header like
```
---
title: featuresets
authors:
  - "@deads2k"
  ...
```
Might be useful here too, but certainly optional.
/lgtm

**Review by soltysh — APPROVED** [review]
/approve

**Pull request merged**